snuba/web/query.py

```diff
@@ -201,26 +233,22 @@ def log_query_and_update_stats(
     timer: Timer,
     stats: MutableMapping[str, Any],
     status: str,
+    query_list: List[ClickhouseQueryMetadata],
     query_settings: Mapping[str, Any],
 ) -> MutableMapping:
     """
     If query logging is enabled then logs details about the query and its status, as
     well as timing information.
     Also updates stats with any relevant information and returns the updated dict.
     """
-
     stats.update(query_settings)
+
     if settings.RECORD_QUERIES:
-        # send to redis
-        state.record_query(
-            {
-                "request": request.body,
-                "sql": sql,
-                "timing": timer,
-                "stats": stats,
-                "status": status,
-            }
+
+        query_metadata = ClickhouseQueryMetadata(
+            request=request.body, sql=sql, timer=timer, stats=stats, status=status
         )
+        query_list.append(query_metadata)
 
         timer.send_metrics_to(
```

Comment:
This would also probably a be a good time to move this function outside of the split inner loop so that we stop recording the accumulated timings from the beginning of the request with each query execution and polluting the metrics.